src/set_theory/ordinal/natural_ops.lean

```diff
@@ -325,4 +338,299 @@ theorem nadd_left_cancel_iff : ∀ {a b c}, a ♯ b = a ♯ c ↔ b = c :=
+  a' ⨳ (b ⨳ c) ♯ a ⨳ (b' ⨳ c) ♯ a ⨳ (b ⨳ c') ♯ a' ⨳ (b' ⨳ c') <
+  a ⨳ (b ⨳ c) ♯ a' ⨳ (b' ⨳ c) ♯ a' ⨳ (b ⨳ c') ♯ a ⨳ (b' ⨳ c') :=
+begin
+  have := nmul_nadd_lt ha (nmul_nadd_lt hb hc),
+  simp only [nmul_nadd, ←nadd_assoc] at this,
+  rwa [nadd_right_comm (a' ⨳ _), nadd_right_comm, nadd_right_comm (a ⨳ _),
+    nadd_right_comm _ (a ⨳ (b' ⨳ c'))] at this
+end
+
+theorem nmul_nadd_le₃' {a' b' c' : ordinal} (ha : a' ≤ a) (hb : b' ≤ b) (hc : c' ≤ c) :
+  a' ⨳ (b ⨳ c) ♯ a ⨳ (b' ⨳ c) ♯ a ⨳ (b ⨳ c') ♯ a' ⨳ (b' ⨳ c') ≤
+  a ⨳ (b ⨳ c) ♯ a' ⨳ (b' ⨳ c) ♯ a' ⨳ (b ⨳ c') ♯ a ⨳ (b' ⨳ c') :=
+begin
+  have := nmul_nadd_le ha (nmul_nadd_le hb hc),
+  simp only [nmul_nadd, ←nadd_assoc] at this,
+  rwa [nadd_right_comm (a' ⨳ _), nadd_right_comm, nadd_right_comm (a ⨳ _),
+    nadd_right_comm _ (a ⨳ (b' ⨳ c'))] at this
+end
+
+theorem lt_nmul_iff₃ : d < a ⨳ b ⨳ c ↔ ∃ (a' < a) (b' < b) (c' < c),
+  d ♯ a' ⨳ b' ⨳ c ♯ a' ⨳ b ⨳ c' ♯ a ⨳ b' ⨳ c' ≤
+  a' ⨳ b ⨳ c ♯ a ⨳ b' ⨳ c ♯ a ⨳ b ⨳ c' ♯ a' ⨳ b' ⨳ c' :=
+begin
+  refine ⟨λ h, _, _⟩,
+  { rcases lt_nmul_iff.1 h with ⟨e, he, c', hc, H₁⟩,
+    rcases lt_nmul_iff.1 he with ⟨a', ha, b', hb, H₂⟩,
+    refine ⟨a', ha, b', hb, c', hc, _⟩,
+    have := nadd_le_nadd H₁ (nmul_nadd_le H₂ hc.le),
+    simp only [nadd_nmul, nadd_assoc] at this,
+    rw [nadd_left_comm, nadd_left_comm d, nadd_left_comm, nadd_le_nadd_iff_left,
+      nadd_left_comm (a ⨳ b' ⨳ c), nadd_left_comm (a' ⨳ b ⨳ c), nadd_left_comm (a ⨳ b ⨳ c'),
+      nadd_le_nadd_iff_left, nadd_left_comm (a ⨳ b ⨳ c'), nadd_left_comm (a ⨳ b ⨳ c')] at this,
+    simpa only [nadd_assoc] },
+  { rintro ⟨a', ha, b', hb, c', hc, h⟩,
+    have := h.trans_lt (nmul_nadd_lt₃ ha hb hc),
+    repeat { rwa nadd_lt_nadd_iff_right at this } }
+end
+
+theorem nmul_le_iff₃ : a ⨳ b ⨳ c ≤ d ↔ ∀ (a' < a) (b' < b) (c' < c),
+  a' ⨳ b ⨳ c ♯ a ⨳ b' ⨳ c ♯ a ⨳ b ⨳ c' ♯ a' ⨳ b' ⨳ c' <
+  d ♯ a' ⨳ b' ⨳ c ♯ a' ⨳ b ⨳ c' ♯ a ⨳ b' ⨳ c' :=
+by { rw ←not_iff_not, simp [lt_nmul_iff₃] }
+
+theorem lt_nmul_iff₃' : d < a ⨳ (b ⨳ c) ↔ ∃ (a' < a) (b' < b) (c' < c),
+  d ♯ a' ⨳ (b' ⨳ c) ♯ a' ⨳ (b ⨳ c') ♯ a ⨳ (b' ⨳ c') ≤
+  a' ⨳ (b ⨳ c) ♯ a ⨳ (b' ⨳ c) ♯ a ⨳ (b ⨳ c') ♯ a' ⨳ (b' ⨳ c') :=
+begin
+  refine ⟨λ h, _, _⟩,
+  { rcases lt_nmul_iff.1 h with ⟨a', ha, e, he, H₁⟩,
+    rcases lt_nmul_iff.1 he with ⟨b', hb, c', hc, H₂⟩,
+    refine ⟨a', ha, b', hb, c', hc, _⟩,
+    have := nadd_le_nadd H₁ (nmul_nadd_le ha.le H₂),
+    simp only [nmul_nadd, nadd_assoc] at this,
+    rw [nadd_left_comm, nadd_left_comm (a' ⨳ (b ⨳ c)), nadd_left_comm (a ⨳ (b ⨳ c')),
+      nadd_left_comm (a ⨳ (b' ⨳ c)), nadd_left_comm (a' ⨳ (b ⨳ c)), nadd_left_comm (a ⨳ e),
+      nadd_le_nadd_iff_left, nadd_left_comm (a' ⨳ (b ⨳ c')), nadd_left_comm (a' ⨳ (b' ⨳ c)),
+      nadd_left_comm d, nadd_le_nadd_iff_left] at this,
+    simpa only [nadd_assoc] },
+  { rintro ⟨a', ha, b', hb, c', hc, h⟩,
+    have := h.trans_lt (nmul_nadd_lt₃' ha hb hc),
+    repeat { rwa nadd_lt_nadd_iff_right at this } }
+end
```

Comment:
```suggestion
theorem nadd_eq_add : a ♯ b = (a.to_nat_ordinal + b.to_nat_ordinal).to_ordinal := rfl

theorem lt_nmul_iff₃' : d < a ⨳ (b ⨳ c) ↔ ∃ (a' < a) (b' < b) (c' < c),
  d ♯ a' ⨳ (b' ⨳ c) ♯ a' ⨳ (b ⨳ c') ♯ a ⨳ (b' ⨳ c') ≤
  a' ⨳ (b ⨳ c) ♯ a ⨳ (b' ⨳ c) ♯ a ⨳ (b ⨳ c') ♯ a' ⨳ (b' ⨳ c') :=
begin
  simp_rw [nmul_comm _ (_ ⨳ _), lt_nmul_iff₃],
  simp only [nadd_eq_add, to_ordinal_to_nat_ordinal], split,
  { rintro ⟨b', hb, c', hc, a', ha, h⟩,
    use [a', ha, b', hb, c', hc], convert h using 1; abel },
  { rintro ⟨a', ha, b', hb, c', hc, h⟩,
    use [b', hb, c', hc, a', ha], convert h using 1; abel },
end
```
Needs `import tactic.abel`. Is `nadd_eq_add` a good name? Are there other golfing opportunity using commutativity?

Reply:
I think `nadd_eq_add` works, I can't think of much further golfing opportunities.